File: src/modernbinary.py
import os
import sys
from core import convert, parser

def main():
    argv = sys.argv[1:]

    if not argv:
        print('[Error] You must enter the file name in the input arguments.')
        sys.exit(1)

    elif(argv[0] == '--texttomb'):
        print(convert.tomb(' '.join(argv[1:])))
        sys.exit(1)

    elif(argv[0] == '--mbtotext'):
        print(convert.totext(' '.join(argv[1:])))
        sys.exit(1)

    else:
        if not argv[0].endswith('.mb'):
            print('[Error] Your file format must be mb.')
            sys.exit(1)

        elif not os.path.isfile(argv[0]):
            print('[Error] There is no file named {}'.format(argv[0]))
            sys.exit(1)

        parser.Parser(argv[0])


if __name__ == '__main__':
    main()

File: src/comp/modernbinary.cpp
#include <iostream>
#include <string>
#include <vector>
#include <stdlib.h>
#include <locale>
#include <cctype>
#include <algorithm>
#include <sstream>
#include "core/convert.cpp"

using namespace std;

static inline string &ltrim(string &s) {
    s.erase(s.begin(), find_if(s.begin(), s.end(),
            not1(ptr_fun<int, int>(isspace))));
    return s;
}

static inline string &rtrim(string &s) {
    s.erase(find_if(s.rbegin(), s.rend(),
            not1(ptr_fun<int, int>(isspace))).base(), s.end());
    return s;
}

static inline string &trim(string &s) {
    return ltrim(rtrim(s));
}

int main(int argc, char *argv[])
{
    string current_exec_name = argv[0];
    vector<string> all_args;
    
    all_args.assign(argv+1, argv + argc);
    argc--;

    if(!argc){
        cout<<"[Error] You must enter the file name in the input arguments."<<endl;
        exit(0);
    }else if(all_args[0] == "--texttomb" || all_args[0] == "--mbtotext"){

        ostringstream stream;
        string arr[all_args.size()];
        int k = 0;
        bool first = false;
        for (string const &i: all_args) {
            if(!first){
                first = true;
                continue;
            }
            arr[k++] = i;
        }
        for (char i = 0; i < sizeof(arr) / sizeof(arr[0]); ++i) {
            if (i) stream << ' ';
            stream << arr[i];
        }
        string to_convert = stream.str();
        to_convert = trim(to_convert);
        string to_output;
        if(all_args[0] == "--texttomb"){
            to_output = texttomb(to_convert);
        }else{
            to_output = mbtotext(to_convert);
        }
        to_output = trim(to_output);
        cout<<to_output<<endl;
        exit(0);
    }else{
        string f_name = all_args[0];
    }
    cout<<endl;
}